### DISCUSSION TITLE
How to execute query to fetch data only when button is pressed

### DISCUSSION THREAD

**Discussion by marcwitteveen** [Power-Components/livewire-powergrid#664] [Q&A]
At this moment, when the page loads with the grid it fetches data immediately. I would like to delay this until a button is pressed.

The idea behind is as followed:

1. Page loads
2. The user creates a filter criteria, where at least 3 fields are set
3. Execute query
4. Grid is populated

**Comment by MichielVervaart** [comment]
I ran into a simular issue, and this is what i came up with:

You just have to implement a button/logic that actually emits the refreshFilters event.

    public function refreshFilters($filters): void
    {
        $this->queryFilters = $filters;
        $this->emitSelf('$refresh');
    }


    public function datasource(): Builder
    {
        if (! $this->queryFilters) {
            return Model::query()->where('id', -1); //empty table
        }

        return Model::query()->ActualQueryStuff();
    }

    protected function getListeners(): array 
    {
        return array_merge(
            parent::getListeners(), [
                'refreshFilters',
            ]);
    }
